### DISCUSSION TITLE
Can I bind a shortcut to "Open in Terminal" action in nemo?

### DISCUSSION THREAD

**Discussion by root2185** [linuxmint/discussions#187] [Questions]
Is there a command to do that?

**Comment by anaximeno** [comment]
This already exists, go to **_Edit > Preferences > Toolbar_** and select the open in toolbar icon, it will show in the toolbar and open the current nemo page in the terminal when clicked.

**Reply by root2185** [reply]
I am looking for a shortcut to do that.
There is also the right click menu.

**Reply by Jeremy7701** [reply]
You could write a nemo action to execute a command or script for selected file types (mime types); You will get a menu entry in the right click menu for those file types it applies to.

**Reply by realroot2185** [reply]
Nemo shows the action already, check the title:
` Can I bind a shortcut to "Open in Terminal" action in nemo?`
Binding a shortcut for that.

**Reply by rcalixte** [reply]
Have you tried the nemo-terminal extension? The main difference is that it's built into the Nemo window/tab rather than a separate terminal window but the extension has an option for toggling the terminal view with a key binding.

**Reply by realroot2185** [reply]
Yes but sometimes you need to open a terminal so that you can do more advanced things.
nemo-terminal is good for simple things.

**Reply by rcalixte** [reply]
Would an alias in the nemo-extension terminal be a good work-around for now? Something like `alias g='gnome-terminal $PWD'` (or whatever terminal you're using) so that you could switch from the extension to a pop-out terminal quickly enough?

The functionality is similar to the toolbar button. If you're using your mouse, the toolbar button is quick enough. If you're already in the terminal but want to open a dedicated new terminal window, the alias would function similar to a key binding (e.g. `g` key and then enter).

**Reply by realroot2185** [reply]
Nice to have but usually I directly open a terminal without using nemo-terminal before, when not doing trivial things.

I now noticed that shortcut closes it and opens it but only the first time.
I always let it open usually.

**Reply by rcalixte** [reply]
> I now noticed that shortcut closes it and opens it but only the first time.

That sounds like a bug. 😔




> Nice to have but usually I directly open a terminal without using nemo-terminal before, when not doing trivial things.

Do you have a context menu button on your keyboard? It's not quite a key binding but being able to open the menu and then navigating to the "Open in Terminal" action also seems like a possible workaround as it keeps you on the keyboard and you can accomplish your desired action without the mouse. (Maybe adding a mnemonic menu option would save you a few key presses too?)

**Reply by rcalixte** [reply]
> > I now noticed that shortcut closes it and opens it but only the first time.
> 
> That sounds like a bug. 😔

Ah, it's not a bug. When the nemo-terminal extension is enabled, the focus switches to the terminal where the key binding toggle for the extension is no longer being sent to Nemo. (In effect, when you enable the terminal pane when it was previously not being shown, the focus shifts directly to the terminal prompt.)

**Reply by realroot2185** [reply]
Yes I have a context menu button, I didn't think about that.
So far I used to right click and "Open in Terminal".
I enabled the button in the toolbar too.

Right If I send the focus back I can use the shortcut.
Changing focus with a keyboard shortcut could be useful too?

When using `ls`/`lsd` folder are shown in dark blue that is pretty hard to see on black background.

**Reply by rcalixte** [reply]
> When using `ls`/`lsd` folder are shown in dark blue that is pretty hard to see on black background.

I don't know if there is a permanent option without inspecting the code, but you can use some variation of `ls --color=none` or `dircolors` (with configurations) that should help? Setting an entire theme could maybe be configured the same way as well but I can't say without spending some time on it.

**Reply by realroot2185** [reply]
I can use `ls --color=none`.

`ls` is set as alias for `ls --color=auto -FAh`, so it shows colors.

**Reply by rcalixte** [reply]
Not to over-complicate things, but you could make a function in your .bashrc file that would set `--color` to either `auto` or `none` for `ls` depending on an environment variable such as `GIO_LAUNCHED_DESKTOP_FILE` if you're certain you always want the colors set a certain kind of way.

If you're okay with it the way it is (for now), that's fine too!

**Reply by realroot2185** [reply]
```
dircolors --print-ls-colors
rs	0
di	01;34

```
It shows always `01;34` for dirs but in nemo-terminal is dark blue: #0000C0.

While in other terminals is light blue: 
tilix #4E6E8F
konsole #3DAEE9
